src/lepton/jpgcoder.cc

```diff
@@ -3531,18 +3531,19 @@ bool recode_jpeg( void )
 
             // (re)set rst wait counter
             rstw = rsti;
-            if (cs_cmpc != colldata.get_num_components() && !g_allow_progressive) {
-                custom_exit(ExitCode::PROGRESSIVE_UNSUPPORTED);
-            }
-            if (jpegtype != 1 && !g_allow_progressive) {
-                custom_exit(ExitCode::PROGRESSIVE_UNSUPPORTED);
-            }
-            if ((jpegtype != 1 || cs_cmpc != colldata.get_num_components())
-                && colldata.is_memory_optimized(0)
-                && first_pass) {
-                colldata.init(cmpnfo, cmpc, mcuh, mcuv, false);
+
+            if (jpegtype != 1 || cs_cmpc != colldata.get_num_components()) {
+                if (! g_allow_progressive) {
+                    fprintf( stderr, "progressive-encoded jpegs aren't supported");
+                    errorlevel.store(2);  // todo: exit progran with ExitCode::PROGRESSIVE_UNSUPPORTED
```

Comment:
What is 2? Please use constant here.

Reply:
What problem is there with leaving here `custom_exit`?

Reply:
1. I copied 2 from similar code chunks. Actually it's exit code of the entire program. I will replace 2 with symbol name in all these chunks later.

2. We should skip over progressive images and continue to process the rest. It's required to check Lepton performance when processing multiple files in the single program run.